backend/src/v5/schemas/tickets/templates.js

```diff
@@ -0,0 +1,152 @@
+	deprecated: defaultFalse,
+	required: defaultFalse,
+	values: Yup.mixed().when(['type'], (val, schema) => {
+		if (val === fieldTypes.MANY_OF || val === fieldTypes.ONE_OF) {
+			return schema.test('Values check', 'Property values must of be an array of values or the name of a preset', (value) => {
+				if (value === undefined) return false;
+				let typeToCheck;
+				if (isString(value)) {
+					typeToCheck = Yup.string().oneOf(Object.values(presetEnumValues)).required();
+				} else {
+					typeToCheck = Yup.array().of(typeNameToType[fieldTypes.ONE_OF]).min(1).required()
+						.strict(true);
+				}
+
+				try {
+					typeToCheck.validateSync(value);
+					return true;
+				} catch (err) {
+					return false;
+				}
+			});
+		}
+		return schema.strip();
+	}),
+
+	default: Yup.mixed().when(['type', 'values'], (type, values) => {
+		const res = typeNameToType[type];
+		if (type === fieldTypes.MANY_OF) {
+			return res.test('Default values check', 'provided values cannot be duplicated and must be one of the values provided', (defaultValues) => {
+				if (defaultValues?.length) {
+					const seenValues = new Set();
+					return defaultValues.every((val) => {
+						const validVal = values.includes(val) && !seenValues.has(val);
+						seenValues.add(val);
+						return validVal;
+					});
+				}
+				return true;
+			});
+		} if (type === fieldTypes.ANY_OF) return res.oneOf(values);
+
+		return res;
+	}),
+
+});
+
+const propertyArray = Yup.array().of(fieldSchema).default([]).test('Property names', 'Property names must be unique inside the same context', (arr) => {
+	const fieldNames = new Set();
+	let res = true;
+	arr.forEach(({ name }) => {
+		const id = name.toUpperCase();
+		if (fieldNames.has(id)) {
+			res = false;
+		} else {
+			fieldNames.add(id);
+		}
+	});
+
+	return res;
+});
+
+const moduleSchema = Yup.object().shape({
+	name: types.strings.title.notOneOf(Object.values(presetModules)),
+	type: Yup.string().oneOf(Object.values(presetModules)),
+	deprecated: defaultFalse,
+	properties: propertyArray.when('type', (type, schema) => {
+		if (type) {
+			const propertiesToCheck = presetModulesProperties[type];
+			return schema.test('No name clash', 'Properties cannot have the same name as a default property',
+				(val) => val.every(({ name }) => !propertiesToCheck.includes(name)));
+		}
+
+		return schema;
+	}),
+}).test('Name and type', 'Only provide a name or a type for a module, not both',
+	({ name, type }) => (name && !type) || (!name && type));
+
+const configSchema = Yup.object().shape({
+	comments: defaultFalse,
+	issueProperties: defaultFalse,
+	defaultView: defaultFalse,
+	defaultImage: Yup.boolean().when('defaultView', (defaultView, schema) => (defaultView ? schema.strip() : defaultFalse)),
+	pin: defaultFalse,
+}).default({});
+
+const defaultPropertyNames = defaultProperties.map(({ name }) => name);
+const schema = Yup.object().shape({
+	name: types.strings.title.required(),
```

Comment:
because this is in the schema folder not the middleware folder

checking something is unique is a very specific situation for things like creating a new ticket or updating a new ticket. 

i want to reuse the same schema for things like casting an output etc, thus checking name being unqiue wouldn't make too much sense here. the schema here is contextless, middeware handles the context

I appreciate this is a bit different to the some of the stuff in the past, but this schema is just too complex and it doesn't seem to make sense to live in middleware when I need it in multiple places.